doc/libpmemobj.3.md

```diff
@@ -1637,26 +1632,81 @@ Using **TX_PARAM_MUTEX** or **TX_PARAM_RWLOCK** means that at the beginning of a
+This mechanism can be deemed as an alternative method for executing code between stages (instead of **TX_ONCOMMIT**,
 **TX_ONABORT**, etc).
-However there are 2 significant differences when nested transactions are used:
 
-+  Registered function is executed only in the most outer transaction (even if registered in the inner one).
+Note that **TX_PARAM_CB** does not replace **TX_ONCOMMIT**/**TX_ONABORT**/etc. macros. They can be used together - a callback will be executed *before* **TX_ONCOMMIT**/**TX_ONABORT**/etc. section.
+
+If **pmemobj_tx_begin**() is called on the same memory pool within an open transaction, it starts a nested transaction on the pool. A new transaction may be started in this way only if the current stage of the open transaction is **TX_STAGE_NONE** or **TX_STAGE_WORK**.
+If successful, the transaction stage changes to **TX_STAGE_WORK** and the function returns zero. Otherwise, thes stage changes to **TX_STAGE_ONABORT** and an error number is returned.
+Transactions nested this way are flattened. Committing the inner transaction does not commit the outer transaction, however errors in the inner transaction are propagated up to the outermost level, resulting in the interruption of the entire transaction.
+
+There are 2 significant differences between nesting transactions and using stage callbacks:
+
++  Registered callback functions are executed only in the most outer transaction (even if registered in the inner one).
 
 +  There can be only one callback in the whole transaction (it can't be changed in the inner transaction).
 
-Note that **TX_PARAM_CB** does not replace **TX_ONCOMMIT**/**TX_ONABORT**/etc. macros. They can be used together - a callback will be executed *before*
-**TX_ONCOMMIT**/**TX_ONABORT**/etc. section.
+**TX_PARAM_CB** can be used when the code dealing with transaction stage changes is shared between multiple users or when it must be executed only in the outer transaction. For example it can be very useful when application must synchronize persistent and transient state.
+
+**pmemobj_tx_begin**() can also be called within an open transaction on a different memory pool. However, for this to succeed the current *transaction context* has to be changed.
+
+When opening the first transaction, a default context is being created. Then all nested transactions on the same pool or consecutive transactions on different pools use the default transaction context until it gets changed explicitly by a user.
+
+The following functions serve for manipulating the transaction context:
+
+```c
+int pmemobj_tx_ctx_set(struct pobj_tx_ctx *new_ctx, struct pobj_tx_ctx **old_ctx);
+int pmemobj_tx_ctx_restore(struct pobj_tx_ctx *ctx);
+```
+
+**pmemobj_tx_ctx_set**() sets the current transaction context to *new_ctx*, storing the previous context under the *old_ctx* pointer. *old_ctx* argument cannot be NULL. Passing NULL for the *new_ctx* allows for creation a new transaction context on the beginning of a subsequent transaction.
```

Comment:
Try minimizing usage of passive voice.
For example, instead of writing:
"allows for creation (of) a new transaction cotext", I'd prefer:
"allows the subsequent transaction to create a context"

Isn't this easier?